### DISCUSSION TITLE
Config file location confusion

### DISCUSSION THREAD

**Discussion by Hr46ph** [siderolabs/talos#7389] [Q&A] [answered]
I created a location for a new deployment in a test environment, ie`mkdir ~/talos-deploy`. I work from that dir, and `talosctl gen config` creates its yaml files there, including `talosconfig` and `kubeconfig`.

Both these config files have default locations, namely `~/.talos/config` and `~/.kube/config`.

Why aren't these files set in their default locations? As far as I am aware, `kubectl` config files can be saved in the `.kube` location which are each parsed individually. I'm not sure about `.talos` though, it would be nice if it did the same.

The docs suggest setting `TALOSCONFIG` variable to the realpath value of the `talosconfig` created with the `gen config` command. Again, I don't understand why `~/.talos/config` isn't used here as well. The whole thing just seems confusing to me.

In addition, I encounter the following issue with certain commands that use cluster config. My `~/talos-deploy/talosconfig` contains my cluster config info:
```
context: talos-cluster
contexts:
    talos-cluster:
        endpoints:
            - ip1
            - ip2
            - ip3
        ca: etc ...
```
Commands such as `talosctl cluster show` and `talosctl reboot` don't seem to work. I can specify the cluster as `talosctl reboot --cluster talos-cluster --wait` or `talosctl cluster show --cluster talos-cluster`, or even specify the config file with `--talosconfig ~/.talos/config`, it seems my commands never accept my cluster name as a parameter. When I use the `--nodes` paramater however, nodes reboot just fine.

I'm not sure if this is my config, or if I'm misunderstanding how cluster parameters work, I am new to Talos so I am just learning as I go. Hence, I post under discussions and not in Issues (although I suspect there is an issue with the cluster config and talos parameters 😉 ).

Any help is appreciated, thanks for reading!

**Comment by frezbo** [comment]
you can use `talosctl config merge <local talosconfig` to get it added to default location.

**Comment by Hr46ph** [comment]
Thanks, that helps.

If someone could look at the "in addition" part, I always have to specify nodes as if my config isn't correct/incomplete or not there. Its not just the cluster show command, its basically any command that's not directly targeted at a specific node, like talosctl get members?

**Reply by frezbo** [reply, marked as answer]
you could set a node/endpoint in the config file as default, eg: `talosctl config node <node ip>`

**Reply by Hr46ph** [reply]
Thank you!

**Comment by Hr46ph** [comment]
Thank you!
